pkg/components/storage.go

```diff
@@ -142,5 +143,9 @@ func startCSIPlugin(ctx context.Context, cfg *config.Config, kubeconfigPath stri
 		klog.Warningf("Failed to apply sccs %v: %v", scc, err)
 		return err
 	}
+	if err := assets.ApplyVolumeSnapshotClass(ctx, kubeconfigPath, vcs, nil, nil); err != nil {
+		klog.Warningf("Failed to apply volumeSnapshotClass %s: %v", vcs, err)
```

Comment:
If returning an error from this function is going to cause the process to exit, let the caller log the message so all of the context is there together at the end of the log.

Reply:
I don't disagree.  The pattern throughout that package is to both drop a warning log and return the err. Maybe we leave it for now and follow up with a PR to correct the bad pattern?